featuretools/tests/primitive_tests/test_feature_serialization.py

```diff
@@ -304,3 +308,154 @@ def test_serialize_url(es):
     error_text = "Writing to URLs is not supported"
     with pytest.raises(ValueError, match=error_text):
         ft.save_features(features_original, URL)
+
+
+def test_deserializer_uses_common_primitive_instances_no_args(es, tmpdir):
+    features = ft.dfs(
+        entityset=es,
+        target_dataframe_name="products",
+        features_only=True,
+        agg_primitives=["sum"],
+        trans_primitives=["is_null"],
+    )
+
+    is_null_features = [f for f in features if f.primitive.name == "is_null"]
+    sum_features = [f for f in features if f.primitive.name == "sum"]
+
+    # Make sure we have multiple features of each type
+    assert len(is_null_features) > 1
+    assert len(sum_features) > 1
+
+    # DFS should use the same primitive instance for all features that share a primitive
+    is_null_primitive = is_null_features[0].primitive
+    sum_primitive = sum_features[0].primitive
+    assert all([f.primitive is is_null_primitive for f in is_null_features])
+    assert all([f.primitive is sum_primitive for f in sum_features])
+
+    file = os.path.join(tmpdir, "features.json")
+    ft.save_features(features, file)
+    deserialized_features = ft.load_features(file)
+    new_is_null_features = [
+        f for f in deserialized_features if f.primitive.name == "is_null"
+    ]
+    new_sum_features = [f for f in deserialized_features if f.primitive.name == "sum"]
+
+    # After deserialization all features that share a primitive should use the same primitive instance
+    new_is_null_primitive = new_is_null_features[0].primitive
+    new_sum_primitive = new_sum_features[0].primitive
+    assert all([f.primitive is new_is_null_primitive for f in new_is_null_features])
+    assert all([f.primitive is new_sum_primitive for f in new_sum_features])
+
+
+def test_deserializer_uses_common_primitive_instances_with_args(es, tmpdir):
```

Comment:
Can start using tmp_path instead of tmpdir:

https://docs.pytest.org/en/latest/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures

Reply:
Updated the two new tests to use `tmp_path` 

[81568d0](https://github.com/alteryx/featuretools/pull/2127/commits/81568d038ddd8ec9d5fad2d457963dbb621c9a4d)